packages/riverpod_generator/lib/src/templates/family.dart

```diff
@@ -229,6 +230,11 @@ ${parameters.map((e) => '        ${e.name}: ${e.name},\n').join()}
     });
 
     final docs = providerDocFor(provider.providerElement.element);
+    var meta = '';
+    final node = provider.node;
+    if (node is AnnotatedNode) {
```

Comment:
Both `ClassDeclaration` and `FunctionDeclaration` are of type `AnnotatedNode`, so theoretically we could write
```dart
final meta = metaAnnotations((provider.node as AnnotatedNode).metadata);
```

Reply:
An "is" check shouldn't be needed then.
Consider updating riverpod_analizer_utils to have "node" be an AnnotatedNode

Reply:
You'd then remove this if condition